packages/ui/v2/core/Shell.tsx

```diff
@@ -648,7 +663,10 @@ function DeploymentInfo() {
 function SideBarContainer() {
   const { status } = useSession();
   const router = useRouter();
-  if (status !== "authenticated") return null;
+  // Make sure that Sidebar is rendered optimistically so that a refresh of pages when logged in have SideBar from the beginning.
+  // This improves the experience of refresh on app store pages(when logged in) which are SSG.
+  // Though when logged out, app store pages would temporarily show SideBar until session status is confirmed.
+  if (status !== "loading" && status !== "authenticated") return null;
```

Comment:
I'll allow it but tagging @Jaibles to have a second opinion

Reply:
BTW, this came from @hariombalhara on his contribution to my other PR merged to this one